hoprd/rest-api/src/session.rs

```diff
@@ -0,0 +1,131 @@
+use std::sync::Arc;
+
+use axum::{
+    extract::{Json, State},
+    http::status::StatusCode,
+    response::IntoResponse,
+};
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, DisplayFromStr};
+
+use hopr_lib::{PathOptions, PeerId, SessionCapability, SessionClientConfig};
+use tokio::{io::copy_bidirectional_with_sizes, net::TcpListener};
+use tokio_util::compat::FuturesAsyncReadCompatExt;
+use tracing::{error, info};
+
+use crate::{ApiErrorStatus, InternalState, BASE_PATH};
+
+#[serde_as]
+#[derive(Debug, Clone, Serialize, Deserialize, utoipa::ToSchema)]
+#[schema(example = json!({
+        "destination": "12D3KooWR4uwjKCDCAY1xsEFB4esuWLF9Q5ijYvCjz5PNkTbnu33",
+        "path": {
+            "Hops": 1
+        },
+        "capabilities": ["Segmentation", "Retransmission"]
+    }))]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct SessionClientRequest {
+    #[serde_as(as = "DisplayFromStr")]
+    pub destination: PeerId,
+    pub path: PathOptions,
+    pub capabilities: Vec<SessionCapability>,
+    pub port: Option<u16>,
+}
+
+impl From<SessionClientRequest> for SessionClientConfig {
+    fn from(value: SessionClientRequest) -> Self {
+        Self {
+            peer: value.destination,
+            path_options: value.path,
+            capabilities: value.capabilities,
+        }
+    }
+}
+
+#[serde_as]
+#[derive(Debug, Clone, Serialize, Deserialize, utoipa::ToSchema)]
+#[schema(example = json!({
+        "port": 5542,
+    }))]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct SessionClientResponse {
+    pub port: u16,
+}
+
+/// Creates a new client session returing a dedicated session listening port.
+#[utoipa::path(
+        get,
+        path = const_format::formatcp!("{BASE_PATH}/session/client"),
+        request_body(
+            content = SessionClientRequest,
+            description = "Configuration of the client session to be created",
+            content_type = "application/json"),
+        responses(
+            (status = 200, description = "Successfully created a new client session", body = SessionClientResponse),
+            (status = 400, description = "Bad request.", body = ApiError),
+            (status = 401, description = "Invalid authorization token.", body = ApiError),
+            (status = 422, description = "Unknown failure", body = ApiError),
+            (status = 500, description = "Internal server error", body = ApiError)
+        ),
+        security(
+            ("api_token" = []),
+            ("bearer_token" = [])
+        ),
+        tag = "Messages"
+    )]
+pub(crate) async fn create_client(
+    State(state): State<Arc<InternalState>>,
+    Json(args): Json<SessionClientRequest>,
+) -> Result<impl IntoResponse, impl IntoResponse> {
+    let port = args.port.unwrap_or(0);
+    let data: SessionClientConfig = args.into();
+    let is_tcp_like = data.capabilities.contains(&SessionCapability::Retransmission)
```

Comment:
This seems overly generic. Either a client asks for a TCP session or not. I would not expect the client to be able handle various options.

Reply:
We could limit the options to a port and forget about the UDP at this point, as we are not using it.